core/src/main/java/com/novoda/merlin/receiver/ConnectivityCallbacks.java

```diff
@@ -0,0 +1,50 @@
+package com.novoda.merlin.receiver;
+
+import android.annotation.TargetApi;
+import android.net.ConnectivityManager;
+import android.net.Network;
+import android.net.NetworkInfo;
+import android.os.Build;
+
+import com.novoda.merlin.service.MerlinService;
+
+@TargetApi(Build.VERSION_CODES.LOLLIPOP)
+public class ConnectivityCallbacks extends ConnectivityManager.NetworkCallback {
+
+    private final ConnectivityManager connectivityManager;
+    private final MerlinService merlinService;
```

Comment:
Oh yeah. That's why I said I'm not aligned :)
As far as I see now, it is not even a problem because `unregister` will be called and this callback will be unregistered. 👍